### PR TITLE
Code clean and basic structure defined

### PR DESCRIPTION
**Pull request by adamryman**
See [README.md] (https://github.com/adamryman/ambition/blob/ground-up/README.md) for code structure details.

I think the code is mostly understandable. I have cut out all the crap and cleaned things up. There still may need comments in some places.

Note there are no tests. I actually tried writing a few but failed. I would like to discuss with @zaqthefreshman about writing tests for what currently exists.

See `/docs/structure.md` for more future details. They are rough though, first draft.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,98 @@
 # ambition
-Event tracking
+
+Action tracking
+
+## Setup
+
+
+Install go [(Instructions here)](https://golang.org/doc/install)
```
Comment (zaqthefreshman) [inline README.md]: most likely this can just be a go get command and you can leave installing go as a first or implied step

File: db.go [hunk 2]
```diff
@@ -0,0 +1,204 @@
+package main
```
Comment (zaqthefreshman) [inline db.go]: recommend moving ambitions logic out into its own package. see https://github.com/zaqthefreshman/ambition/tree/zaq_skeleton/cmd/main.go for an example of a broken out entry point.

File: README.md (hunk 1 above)
Comment (adamryman) [inline README.md]: Good suggestion, I like it.

File: db.go [hunk 3]
```diff
@@ -0,0 +1,204 @@
...
+	for rows.Next() {
+		var occurrence Occurrence
+		err := rows.Scan(&occurrence.Id, &occurrence.ActionId, &occurrence.Time)
+		check(err)
+		reval = append(reval, occurrence)
+	}
+	return reval, err
+}
+
+func (db DB) InsertOccurrence(occurrence *Occurrence) error {
```
Comment (zaqthefreshman) [inline db.go]: recommend wrapping these statements in a transaction. Possibly outside the scope of this PR, defs opens an issues if we decide to skip though. http://golang.org/pkg/database/sql/#DB.Begin

File: docs/structure.md [hunk 4]
```diff
@@ -0,0 +1,212 @@
+# ambition
+
+ambition is an action tracker for the purpose of quantitative self and habit tracking.
+
+The core will be database with a useful structure and an api that can be accessed by many different interfaces for both data input and viewing.
+
+## Note
```
Comment (zaqthefreshman) [inline docs/structure.md]: I would remove or rephrase this note, people will do it anyways and it takes away from your project.

File: docs/structure.md [hunk 5]
```diff
@@ -0,0 +1,212 @@
...
+
+This is a learning project, please point out anything that is dumb or that could be done better.
+
+## Table structure
+
+Users have Sets, Sets have Actions, and Actions have Occurrences.
+
+A user would be person (or a group? What about groups?) A set would be a logical group of actions. An action is an thing that can be done, binary in completion. An occurrences is an instance of an action being done. An occurrence may have metadata. The action determines whether their occurrences will have metadata, the metadata structure, and whether the metadata is optional or not.
```
Comment (zaqthefreshman) [inline docs/structure.md]: spelling and grammar should be addressed now. Later own we should aim to simplify this doc as well.

File: main.go [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+		// Add the routes in route.go
+		AddRoutes(router)
+
+		// Start the http server
+		http.ListenAndServe(":3000", router)
+	}
+}
+
+// Code that can be run once at the start of ambition. To be removed
+func oneOff() {
```
Comment (zaqthefreshman) [inline main.go]: wat

File: main.go [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+var _commands = map[string]func(){
+	"seed":   database.seedTables,
+	"create": database.createTables,
+	"drop":   database.dropTables,
+}
+
+func main() {
+	// database located in db.go
+	defer database.Close()
+	oneOff()
```
Comment (zaqthefreshman) [inline main.go]: wat

**Comment by zaqthefreshman** [comment]
Rebase!

**Comment by zaqthefreshman** [comment]
Consider moving you seed data into a some kind of testdata folder. Since it mostly looks like something that would be consumed by unittests in a mocked database moving forward.

File: config/occurrences-seed.json [hunk 8]
```diff
@@ -0,0 +1,27 @@
+[
+			{
```
Comment (zaqthefreshman) [inline config/occurrences-seed.json]: tabbing

File: db.go [hunk 9]
```diff
@@ -0,0 +1,204 @@
...
+)
+
+// Database type to extend with custom functions
+type DB struct {
+	*sql.DB
+}
+
+// Database initialization
+// TODO: Add seting up postgresql
+var tempdb, _ = sql.Open("postgres", "dbname=ambition user=ambition password=ambition sslmode=disable")
```
Comment (zaqthefreshman) [inline db.go]: could easily be in an env var

**Comment by zaqthefreshman** [comment]
LGTM :shipit:

**Pull request merged**